File: internal/content/testdata/bench/javascript/nc-sliding-window-max.js
function max_sliding_window(nums, k) {
    const n = nums.length;
    const result = [];
    const dq = []; // indices, decreasing by value
    for (let i = 0; i < n; i++) {
        while (dq.length > 0 && dq[0] < i - k + 1) {
            dq.shift();
        }
        while (dq.length > 0 && nums[dq[dq.length - 1]] < nums[i]) {
            dq.pop();
        }
        dq.push(i);
        if (i >= k - 1) {
            result.push(nums[dq[0]]);
        }
    }
    return result;
}

File: internal/content/testdata/bench/go/nc-sliding-window-max.go
func max_sliding_window(nums []int, k int) []int {
	n := len(nums)
	result := []int{}
	dq := []int{} // indices, decreasing by value
	for i := 0; i < n; i++ {
		for len(dq) > 0 && dq[0] < i-k+1 {
			dq = dq[1:]
		}
		for len(dq) > 0 && nums[dq[len(dq)-1]] < nums[i] {
			dq = dq[:len(dq)-1]
		}
		dq = append(dq, i)
		if i >= k-1 {
			result = append(result, nums[dq[0]])
		}
	}
	return result
}
